tools/slicec-cs/src/main.rs

```diff
@@ -34,7 +35,7 @@ pub fn main() {
     let slice_options = &cs_options.slice_options;
 
     let mut compilation_state = slicec::compile_from_options(slice_options, cs_patcher, cs_validator);
-
+    let hash = SliceFile::compute_sha256_hash(&compilation_state.files);
```

Comment:
Could we move this down into the `cs_options.telemetry` if-check?
1) That's the only place where it's used, and it's better to put code where it's used
2) If it telemetry isn't enabled, we shouldn't bother computing it (since we do nothing with it).

I'd also like to keep the space between `let mut compilation_state` and the `if !compilation_state`.
These are both very big steps of compiler execution, and don't really relate to each other directly.